apps/simple-trading-app/src/app/components/simple-market-list/data-provider.ts

```diff
@@ -87,15 +88,16 @@ export const FILTERS_QUERY = gql`
 `;
 
 const update = (
-  draft: SimpleMarkets_markets[],
+  data: SimpleMarkets_markets[],
   delta: SimpleMarketDataSub_marketData
-) => {
-  const index = draft.findIndex((m) => m.id === delta.market.id);
-  if (index !== -1) {
-    draft[index].data = delta;
-  }
-  // @TODO - else push new market to draft
-};
+) =>
+  produce(data, (draft) => {
```

Comment:
question (non-blocking) Dont we have to use the result of the produce call here?

Reply:
I didn't see the implicit return from the arrow function. For the record I prefer to only do this for one liners. Can we add some curly braces and add a explicit return statement?